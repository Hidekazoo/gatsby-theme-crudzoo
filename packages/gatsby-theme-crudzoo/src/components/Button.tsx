import cn from "classnames"
import React from "react"

import styles from "../styles/components/Button.module.css"

interface ButtonProps {
  type?: "primary"
}
export const Button: React.FC<ButtonProps> = props => {
  const { type = "primary", children, ...rest } = props
  return (
    <button
      className={cn(styles.btn, {
        [styles.primary]: type === "primary",
      })}
      {...rest}
    >
      {children}
    </button>
  )
}
