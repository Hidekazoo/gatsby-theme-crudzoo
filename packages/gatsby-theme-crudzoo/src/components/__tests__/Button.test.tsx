import React from "react"
import { render } from "@testing-library/react"

import { Button } from "../Button"

describe("Button", () => {
  test("Button renders correctly", () => {
    const { container } = render(<Button>Button</Button>)
    expect(container).toMatchSnapshot()
  })
})
