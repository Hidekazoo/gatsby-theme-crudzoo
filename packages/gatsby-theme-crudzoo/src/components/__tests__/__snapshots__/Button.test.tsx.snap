// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Button renders correctly 1`] = `
<div>
  <button
    class="btn primary"
  >
    Button
  </button>
</div>
`;
